Add documentation of upgrading and downgrading (#962)
jellyfin/jellyfin.org: docs/general/testing/upgrades.md

Reviewer: Please increase the numbering on the numbered lists.

Author: I'm not sure what you mean by this, the linter complains if I use `1., 2., 3.` instead of `1., 1., 1.`.

Reviewer 2: I think it is because you have empty lines between each item in the list and the linter is treating it as multiple numbered lists.

---
uid: upgrades-and-downgrades
title: Upgrades & Downgrades
description: Upgrade and downgrade to and from Unstable releases
sidebar_position: 2
---

This document provides details on upgrading and downgrading between Stable and Unstable releases, and why you might want to do so.

## Stable vs. Unstable

Which install type to pick depends on your needs; ultimately, Unstable is for testing new things, while Stable is for running a server for others to use reliably.

* Stable provides the most consistent and predictable user experience. A particular major release (e.g. 10.8.z, 10.9.z) will not introduce, remove, or change new major features or changes (with minor caveats for security). Bugfixes are provided in point releases, which are released as needed during the lifecycle of the major release in response to bugfixes and security advisories. Most users should generally use Stable releases, as this will ensure maximum uptime and consistency for your end users.

> * Unstable provides the most up-to-date, cutting edge features, but may have rapid and unpredictable breaking changes or serious bugs, and have much more limited client support as API changes are made over time. Unstable releases are provided via weekly binary builds on Monday mornings around 5AM UTC, or by [building your own packages from the `master` code branches](https://github.com/jellyfin/jellyfin-packaging).

Author: Tweaked this wording in a slightly different way.

**Always make a backup of your instance data and configuration directories before upgrading to Unstable for any reason.** You never know what might go wrong, be it a failure in migrations, a breaking bug that forces a downgrade back, or some other corruption. This applies even for testing right before a Stable release. It is always better to have the backup ready, perform some testing immediately after upgrading, and revert if things aren't working right - after reporting your bug, of course!

You can utilize certain plugins to help with this, for instance the Trakt plugin to synchronize watched status, but this will not necessarily preserve everything. So be careful here, and know the limitations before proceeding.

How to back up your server depends on the exact platform, as each platform stores data in different places. [The exact locations are documented here](/docs/general/administration/configuration/#server-paths).

**Ensure you stop Jellyfin from running before taking a backup**. If you don't, the database might still be locked and this can cause problems when restoring.

### Disable Automatic Updates on Unstable

> Running with automatic updates while on Unstable, or during a pre-release testing of Unstable, can cause problems. Always ensure you update manually and test things out afterwards. We generally recommend against automatic updates *in general* even on Stable, as it can result in missed release notes from new versions, but we understand how desirable this can be; avoid the temptation when running Unstable though to avoid missing important changes.

Author: There's a multitude of ways and we don't document how to turn it on, so, best to leave this vague.

## Upgrading from Stable to Unstable

> The exact details of this process depend on your platform and installation method. This document will detail the two most popular: Debian/Ubuntu packages via our repository, and Docker. The process is generally the same for others (e.g. manual downloads, Windows installers, etc.) with the exact details changed; for the specifics of individual platforms, please see the child articles under ["Testing Jellyfin Server"](/docs/general/testing/server).

Reviewer 2: ```suggestion
The exact details of this process depend on your platform and installation method. This document will detail the two most popular: Debian/Ubuntu packages via our repository, and Docker. The process is generally the same for others (e.g. manual downloads, Windows installers, etc.) with the exact details changed; for the specifics of individual platforms, please see the child articles under [Testing Jellyfin Server](/docs/general/testing/server).
```

   ```sh
   https://repo.jellyfin.org/files/plugin/manifest.json
   ```

### Debian/Ubuntu (Unstable to Release)

1. Edit your `/etc/apt/sources.list.d/jellyfin.sources` and remove `unstable` from the `Components:` line.

   ```sh
   ...
   Components: main
   ...
   ```

1. Run `sudo apt-get update`.

1. Run `sudo apt-get install --reinstall jellyfin jellyfin-server jellyfin-web`. This will forcibly reinstall the new Stable version over top of the Unstable version. Jellyfin will automatically restart.

### Docker (Unstable to Release)

1. Stop your Jellyfin container.

1. Pull the `jellyfin/jellyfin:latest` image tag.

1. Start a new Jellyfin container with the updated image.

### Post-Install (Unstable to Release)

1. Once updated plugins become available, Jellyfin should automatically upgrade these to the next stable version as mentioned above. If you find any that do not, try to upgrade them manually.

## Downgrading from Unstable to an older Stable

As mentioned above, this process requires restoring from a backup taken before you first upgraded to Unstable.

> The exact details of this process depend on your platform and installation method. This document will detail the two most popular: Debian/Ubuntu packages via our repository, and Docker. The process is generally the same for others (e.g. manual downloads, Windows installers, etc.) with the exact details changed; for the specifics of individual platforms, please see the child articles under ["Testing Jellyfin Server"](/docs/general/testing/server).

Reviewer 2: ```suggestion
The exact details of this process depend on your platform and installation method. This document will detail the two most popular: Debian/Ubuntu packages via our repository, and Docker. The process is generally the same for others (e.g. manual downloads, Windows installers, etc.) with the exact details changed; for the specifics of individual platforms, please see the child articles under [Testing Jellyfin Server](/docs/general/testing/server).
```

We assume you are already running Jellyfin Unstable releases.

### Debian/Ubuntu (Unstable to Old Stable)

1. Stop Jellyfin using the service manager e.g. `sudo service jellyfin stop` or `sudo systemctl stop jellyfin`. Ensure Jellyfin has actually stopped.

1. Restore your backup of the configuration and data directories. Ensure you remove the current contents entirely (or move it out of the way) first.

1. Edit your `/etc/apt/sources.list.d/jellyfin.sources` and remove `unstable` from the `Components:` line.

   ```sh
   ...
   Components: main
   ...
   ```

1. Run `sudo apt-get update`.

> 1. Run `sudo apt-get install --reinstall jellyfin jellyfin-server jellyfin-web`. This will forcibly reinstall the new Stable version over top of the Unstable version. Jellyfin should automatically start; if not, use the service manager e.g. `sudo service jellyfin start` or `sudo systemctl start jellyfin`.

Reviewer 2: ```suggestion
1. Run `sudo apt-get install --reinstall jellyfin jellyfin-server jellyfin-web`. This will forcibly reinstall the new Stable version over top of the Unstable version. Jellyfin should automatically start; if not, `sudo service jellyfin start` or `sudo systemctl start jellyfin` to start Jellyfin.
```
This command should be the same on all debian / ubuntu distros except the crazy ones that remove systemd

Author: How about:

>use the service manager to start Jellyfin (`sudo service jellyfin start` or similar).

Reviewer 2: that should work too
